### ISSUE TITLE
bootctl set-default: fall back to loader.conf?

### ISSUE THREAD

**Issue by lnussel** [systemd/systemd#30705]
### Component

bootctl

### Is your feature request related to a problem? Please describe

On e.g. a rpi u-boot is used to provide an EFI emulation. That's good enough to boot. However doesn't seem to support writing EFI variables as required eg for set-default. Maybe bootctl should fall back to writing loader.conf if the EFI variable can't be updated?

**Comment by bluca** [comment]
IIRC the issue with uboot is not that it doesn't support variables, is that it doesn't set up storage by default, and it has to be explicitly configured? Could you just set up storage on the ESP?

**Comment by michaelolbrich** [comment]
No. As soon as Linux is running, uboot can no longer access most of the hardware as that would conflict with Linux. So it cannot write anything directly. There are ways to work around this but they are quite complex. It's not something you can expect to be always available.

**Comment by bluca** [comment]
Sure but it doesn't need to, some uboot userspace utilities would do that, just like any other implementation does
